### PR TITLE
gl: add actor engine impl for graphlearn-training

### PR DESCRIPTION
**Pull request by goldenleaves**
1. The new server impl is based on hiactor, using this impl will override the default thead pool impl.
2. Configure with CMake option `-DWITH_HIACTOR=ON` to build with hiactor.
3. Set the global flag with`SetGlobalFlagEnableActor(1);` to enable actor engine.

### CODE REVIEW

File: graphlearn/src/actor/operator/full_sampler.act.cc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace act {
+
+FullSamplerActor::FullSamplerActor(hiactor::actor_base* exec_ctx,
+                                   const hiactor::byte_t* addr)
+    : BaseOperatorActor(exec_ctx, addr) {
+  set_max_concurrency(UINT32_MAX);  // stateless
+  SetOp("FullSampler");
+  auto& tm = GetParams();
+  edge_type_ = tm.at(kEdgeType).GetString(0);
+  sampling_strategy_ = tm.at(kStrategy).GetString(0);
```
Comment (Seventeen17) [inline graphlearn/src/actor/operator/full_sampler.act.cc:30]: Please help to add the following code line:
```C++
neighbor_count_ = tm.at(kNeighborCount).GetInt32(0);
```
for supporting the truncated neighbor size in the latest version of `FullSampler`.

File: graphlearn/src/actor/operator/full_sampler.act.cc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+FullSamplerActor::~FullSamplerActor() = default;
+
+seastar::future<TensorMap> FullSamplerActor::Process(TensorMap&& tensors) {
+  // create request
+  SamplingRequest request(edge_type_, sampling_strategy_, 0);
```
Comment (Seventeen17) [inline graphlearn/src/actor/operator/full_sampler.act.cc:37]: As mentioned above, please update it with
```
SamplingRequest request(edge_type_, sampling_strategy_, neighbor_count_, 0);
```

File: graphlearn/src/actor/operator/full_sampler.act.cc (hunk 1 above)
Comment (baoleai) [reply graphlearn/src/actor/operator/full_sampler.act.cc:30]: We need to consider how to better keep the actor and non-actor versions in sync

File: graphlearn/src/actor/operator/full_sampler.act.cc (hunk 2 above)
Comment (goldenleaves) [reply graphlearn/src/actor/operator/full_sampler.act.cc]: fixed

File: graphlearn/src/actor/operator/full_sampler.act.cc (hunk 1 above)
Comment (goldenleaves) [reply graphlearn/src/actor/operator/full_sampler.act.cc:30]: fixed

**Review by Seventeen17 — APPROVED** [review]

**Review by baoleai — APPROVED** [review]

**Pull request merged**